addons/pos_loyalty/models/pos_config.py

```diff
@@ -87,22 +87,26 @@ def use_coupon_code(self, code, creation_date, partner_id):
                 },
             }
         check_date = fields.Date.from_string(creation_date[:11])
+        today_date = fields.Date.context_today(self)
+        error_message = False
         if (coupon.expiration_date and coupon.expiration_date < check_date) or\
-            (coupon.program_id.date_to and coupon.program_id.date_to < fields.Date.context_today(self)) or\
+            (coupon.program_id.date_to and coupon.program_id.date_to < today_date) or\
             (coupon.program_id.limit_usage and coupon.program_id.total_order_count >= coupon.program_id.max_usage):
+            error_message = _('This coupon is expired (%s).', code)
```

Comment:
```suggestion
            error_message = _("This coupon is expired (%s).", code)
```
We follow the guideline of single-quoted strings for technical values, and double-quoted strings for strings shown to the user ;).